VideoWeb/VideoWeb/ClientApp/src/app/shared/mic-visualiser/mic-visualiser.component.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { ChangeDetectorRef } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+import { MicVisualiserComponent } from './mic-visualiser.component';
+
+describe('MicVisualiserComponent', () => {
+    let component: MicVisualiserComponent;
+    let fixture: ComponentFixture<MicVisualiserComponent>;
+    let changeDetectorSpy: jasmine.SpyObj<ChangeDetectorRef>;
+    let canvasElement;
+    let canvasContext;
+
+    beforeEach(async () => {
+        changeDetectorSpy = jasmine.createSpyObj<ChangeDetectorRef>('ChangeDetectorRef', ['detectChanges']);
+
+        await TestBed.configureTestingModule({
+            declarations: [MicVisualiserComponent],
+            providers: [{ provide: ChangeDetectorRef, useValue: changeDetectorSpy }]
+        }).compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(MicVisualiserComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+
+        canvasElement = fixture.debugElement.query(By.css('#meter')).nativeElement;
+        canvasContext = canvasElement.getContext('2d');
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should call setupStream when there is a change', () => {
+        spyOn(component, 'setupStream');
+        component.ngOnChanges();
+        expect(component.setupStream).toHaveBeenCalledTimes(1);
+    });
+
+    it('should setup canvas afterViewInit', () => {
+        component.ngAfterViewInit();
+        fixture.detectChanges();
+        console.log(canvasContext);
```

Comment:
remove this console log unless its been left intentionally?

Reply:
Done